lib/src/request.dart

```diff
@@ -56,7 +58,10 @@ class Request extends BaseRequest {
     _defaultEncoding = value;
     var contentType = _contentType;
     if (contentType == null) return;
-    _contentType = contentType.change(parameters: {'charset': value.name});
+    if (isIgnoreEncoding)
+      _contentType = contentType.change(parameters: {});
```

Comment:
Any `if` with an `else` needs both blocks in braces.

https://www.dartlang.org/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures